llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -3838,6 +3861,11 @@ FixedScalableVFPair LoopVectorizationCostModel::computeFeasibleMaxVF(
   if (!Legal->isSafeForAnyVectorWidth())
     this->MaxSafeElements = MaxSafeElements;
 
+  if (MaxSafeScalableVF != ElementCount::getScalable(0)) {
+    // Cache the value of vscale for tuning, since we'll need it.
+    initializeVScaleForTuning();
```

Comment:
IIUC there's no nice sentinel value we could use to detect calling `getVScaleForTuning` before initialization.  Would be good to either detect calling before initialization or possibly initialize in constructor?

Reply:
I'm not sure I follow. Are you worried about us initialising VScaleForTuning twice? I initially avoided doing this in the constructor because I thought then we'd be doing unnecessary work if the target doesn't support scalable vectors. However, I guess the attribute won't be present in that case so the check is cheap. I'm happy to move into a constructor in that case.

Reply:
> I'm not sure I follow. Are you worried about us initialising VScaleForTuning twice

More about calling it before it was initialized, thanks for the update